api/src/main/java/xtdb/types/IntervalDayTime.java

```diff
@@ -9,6 +9,8 @@ public final class IntervalDayTime {
     public final Duration duration;
 
     public IntervalDayTime(Period period, Duration duration) {
+        if ( period.getYears() != 0 || period.getMonths() != 0)
+            throw new java.lang.IllegalArgumentException("Period can not contain years or months!");
```

Comment:
you don't need to fully qualify `java.lang` in Java, it's auto-imported. also, prefer to use our `xtdb.IllegalArgumentException` (extends j.l.IAE) so that it gets correctly serialised over the wire